src/PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.cpp

```diff
@@ -227,38 +226,61 @@ tnsr::abb<DataType, Dim, Frame> derivatives_of_spacetime_metric(
     const tnsr::ii<DataType, Dim, Frame>& spatial_metric,
     const tnsr::ii<DataType, Dim, Frame>& dt_spatial_metric,
     const tnsr::ijj<DataType, Dim, Frame>& deriv_spatial_metric) noexcept {
-  auto spacetime_deriv_spacetime_metric =
-      make_with_value<tnsr::abb<DataType, Dim, Frame>>(lapse, 0.);
+  tnsr::abb<DataType, Dim, Frame> spacetime_deriv_spacetime_metric{};
+  derivatives_of_spacetime_metric(
+      make_not_null(&spacetime_deriv_spacetime_metric), lapse, dt_lapse,
+      deriv_lapse, shift, dt_shift, deriv_shift, spatial_metric,
+      dt_spatial_metric, deriv_spatial_metric);
+  return spacetime_deriv_spacetime_metric;
+}
 
-  get<0, 0, 0>(spacetime_deriv_spacetime_metric) =
+template <size_t Dim, typename Frame, typename DataType>
+void derivatives_of_spacetime_metric(
+    const gsl::not_null<tnsr::abb<DataType, Dim, Frame>*>
+        spacetime_deriv_spacetime_metric,
+    const Scalar<DataType>& lapse, const Scalar<DataType>& dt_lapse,
+    const tnsr::i<DataType, Dim, Frame>& deriv_lapse,
+    const tnsr::I<DataType, Dim, Frame>& shift,
+    const tnsr::I<DataType, Dim, Frame>& dt_shift,
+    const tnsr::iJ<DataType, Dim, Frame>& deriv_shift,
+    const tnsr::ii<DataType, Dim, Frame>& spatial_metric,
+    const tnsr::ii<DataType, Dim, Frame>& dt_spatial_metric,
+    const tnsr::ijj<DataType, Dim, Frame>& deriv_spatial_metric) noexcept {
+  destructive_resize_components(spacetime_deriv_spacetime_metric,
+                                get_size(get(lapse)));
+  for (auto& component : *spacetime_deriv_spacetime_metric) {
```

Comment:
looks like the only components that need to be zero initially are `spacetime_deriv_spacetime_metric(a,0,i)` where `a` is all dim+1 spacetime components, and `i` is just the dim spatial components